src/generator.cs

```diff
@@ -7006,10 +7053,11 @@ public void Generate (Type type)
 							} else
 								eaname = "<NOTREACHED>";
 							
-							if (bta.Singleton || mi.GetParameters ().Length == 1)
-								print ("EventHandler handler = {0};", PascalCase (miname));
-							else
-								print ("EventHandler<{0}> handler = {1};", GetEventArgName (mi), miname);
+							print ("var handler = {0};", PascalCase (miname));
+							// if (bta.Singleton || mi.GetParameters ().Length == 1)
```

Comment:
Commented out?

Reply:
yup, forgot to remove it. It's now using `var`, which is simpler than detecting the nullability requirement.